resonate/scheduler.py

```diff
@@ -895,6 +896,11 @@ def _eval(self, node: Node[State]) -> list[Function | Delayed[Function | Retry]
                     case TRM(id, result), _:
                         assert id == node.id, "Id must match node id."
 
+                        match result:
+                            case Ok(v) if opts.validation is not None and not opts.validation(v):
+                                msg = "function result didn't passed validation"
+                                result = Ko(ResonateValidationError(msg))
```

Comment:
Can we use a simple `ValueError`? I recently swapped out our custom validation error for a `ValueError` everywhere else.